lightning/src/ln/channel.rs

```diff
@@ -2863,22 +2863,11 @@ impl<SP: Deref> Channel<SP> where
 	/// If we do fail twice, we `debug_assert!(false)` and return `Ok(None)`. Thus, this will always
 	/// return `Ok(_)` if preconditions are met. In any case, `Err`s will only be
 	/// [`ChannelError::Ignore`].
-	pub fn queue_fail_htlc<L: Deref>(&mut self, htlc_id_arg: u64, err_packet: msgs::OnionErrorPacket, logger: &L)
-	-> Result<(), ChannelError> where L::Target: Logger {
-		self.fail_htlc(htlc_id_arg, err_packet, true, logger)
-			.map(|msg_opt| assert!(msg_opt.is_none(), "We forced holding cell?"))
-	}
-
-	/// Used for failing back with [`msgs::UpdateFailMalformedHTLC`]. For now, this is used when we
-	/// want to fail blinded HTLCs where we are not the intro node.
-	///
-	/// See [`Self::queue_fail_htlc`] for more info.
-	pub fn queue_fail_malformed_htlc<L: Deref>(
```

Comment:
To get rid of this method, I had to increase the visibility of some internal `Channel` structs. Might be better to just keep a separate method for failing malformed HTLCs, let me know if there are thoughts.